p2p/net/swarm/swarm.go

```diff
@@ -961,3 +961,31 @@ func (r ResolverFromMaDNS) ResolveDNSComponent(ctx context.Context, maddr ma.Mul
 	}
 	return addrs, nil
 }
+
+// AddCertHashes adds certificate hashes to relevant transport addresses, if there
+// are no certhashes already present on the method. It mutates `listenAddrs`.
+// This method is useful for adding certhashes to public addresses discovered
+// via identify, nat mapping, or provided by the user.
```

Comment:
This comment needs some editing. Maybe:

```suggestion
// AddCertHashes mutates listenAddrs. It adds applicable certificate hashes to
// the multiaddrs. It will not modify a multiaddr if it already contains a
// certificate hash. This method is useful for adding certhashes to public
// addresses discovered via identify, nat mapping, or provided by the user.
```

Reply:
Missed this. I'll open a separate PR.